src/MOI/MOI_wrapper.jl

```diff
@@ -2009,8 +2009,10 @@ function MOI.get(model::Optimizer, attr::MOI.DualStatus)
     if primal_stat == dual_stat == 1
         return MOI.FEASIBLE_POINT
     elseif primal_stat == 0 && dual_stat == 1
-        @assert MOI.get(model, MOI.TerminationStatus()) == MOI.INFEASIBLE
-        return MOI.INFEASIBILITY_CERTIFICATE
+        if MOI.get(model, MOI.TerminationStatus()) == MOI.INFEASIBLE
+            return MOI.INFEASIBILITY_CERTIFICATE
+        end
+        return MOI.UNKNOWN_RESULT_STATUS
```

Comment:
I guess the other option here is to not return and exit at the bottom of the function with `MOI.NO_SOLUTION`. Do you have a preference?

Reply:
It depends on what CPLEX is returning as dual solution. Is it garbase/NaN/empty vector, then it should be `NO_SOLUTION`. If it is the current iterate but CPLEX is not certain what this certifies then it should be `UNKNOWN_RESULT_STATUS`.

Reply:
Hmmm... I'm going to have issues recreating it. The issue happened when I was debugging a larger routine and was since resolved. Right now, the `MOI.OTHER_ERROR` is a catch all for any unsupported solution status so it's not clear to me why is occurred. Maybe this should be updated first then we update the `DualStatus` to return a value based on the more descriptive termination status.

Proposal below:

- We could enumerate all the listed status codes. It looks like the following are missing: 13, 14, 15, 16, 17, 18, 19, 20, 23, 24, 30, 31, 32, 35, 37, 38, 40, 41 and a number of MIP status codes. A lot of the status codes now have more descriptive `TerminationStatusCodes` descriptors.
- A `stat == 0` would result in `MOI.OTHER_ERROR`.
- A `stat` value greater than 0 and not enumerated would throw an error of the style: "getstat() returned at value of $stat which MOI cannot interpret. Please open an issue here to add support for this error code https://github.com/JuliaOpt/CPLEX.jl/issues."

Current routine for reference:
```julia
function MOI.get(model::Optimizer, attr::MOI.TerminationStatus)
    _throw_if_optimize_in_progress(model, attr)
    if model.cached_solution === nothing
        return MOI.OPTIMIZE_NOT_CALLED
    end
    stat = c_api_getstat(model.inner)
    if stat == 1 # CPX_STAT_OPTIMAL
        return MOI.OPTIMAL
    elseif stat == 3 # CPX_STAT_INFEASIBLE
        return MOI.INFEASIBLE
    elseif stat == 4 # CPX_STAT_INForUNBD
        return MOI.INFEASIBLE_OR_UNBOUNDED
    elseif stat == 2 # CPX_STAT_UNBOUNDED
        return MOI.DUAL_INFEASIBLE
    elseif stat in (12, 21, 22, 36) # CPX_STAT_*ABORT*_OBJ_LIM
        return MOI.OBJECTIVE_LIMIT
    elseif stat in (10, 34) # CPX_STAT_*ABORT_IT_LIM
        return MOI.ITERATION_LIMIT
    elseif stat == 53 # CPX_STAT_CONFLICT_ABORT_NODE_LIM
        return MOI.NODE_LIMIT
    elseif stat in (11, 25, 33, 39) # CPX_STAT_*ABORT*TIME_LIM
        return MOI.TIME_LIMIT
    elseif stat == 5 # CPX_STAT_OPTIMAL_INFEAS
        return MOI.NUMERICAL_ERROR
    # MIP STATUS
    elseif stat in (101, 102) # CPXMIP_OPTIMAL, CPXMIP_OPTIMAL_TOL
        return MOI.OPTIMAL
    elseif stat == 103 # CPXMIP_INFEASIBLE
        return MOI.INFEASIBLE
    elseif stat == 119 # CPXMIP_INForUNBD
        return MOI.INFEASIBLE_OR_UNBOUNDED
    elseif stat == 118 # CPXMIP_UNBOUNDED
        return MOI.DUAL_INFEASIBLE
    elseif stat in (105, 106) # CPXMIP_NODE_LIM*
        return MOI.NODE_LIMIT
    elseif stat in (107, 108, 131, 132) # CPXMIP_*TIME_LIM*
        return MOI.TIME_LIMIT
    else
        return MOI.OTHER_ERROR
    end
end
```